solidity/contracts/AcreMultiAssetVault.sol

```diff
@@ -0,0 +1,278 @@
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function addSupportedAsset(address asset) external onlyOwner {
+        return _addSupportedAsset(asset);
+    }
+
+    /// @notice Adds an asset to the list of assets supported by the vault.
+    /// @param asset Address of the asset to be added to the supported assets.
+    function _addSupportedAsset(address asset) internal {
+        if (asset == address(0)) {
+            revert ZeroAddress();
+        }
+        if (supportedAssets[asset]) {
+            revert AssetAlreadySupported();
+        }
+
+        emit SupportedAssetAdded(asset);
+
+        supportedAssets[asset] = true;
+    }
+
+    /// @notice Removes an asset from the list of assets supported by the vault.
+    /// @dev This function can only be called by the owner.
+    /// @param asset Address of the asset to be removed from the supported assets.
+    function removeSupportedAsset(address asset) external onlyOwner {
+        if (!supportedAssets[asset]) {
+            revert AssetNotSupported();
+        }
+
+        emit SupportedAssetRemoved(asset);
+
+        supportedAssets[asset] = false;
+    }
+
+    /// @notice Deposits an asset into the vault.
+    /// @param asset Address of the asset to be deposited.
+    /// @param amount Amount of the asset to be deposited (in the asset precision).
+    /// @return The ID of the deposit.
+    function deposit(address asset, uint256 amount) external returns (uint256) {
+        return depositFor(asset, amount, msg.sender);
+    }
+
+    /// @notice Deposits assets into the vault on behalf of a specified account.
+    /// @param asset Address of the asset to be deposited.
+    /// @param amount The amount of assets to deposit.
+    /// @param depositOwner The address of the account for which the deposit is
+    ///        being made.
+    /// @return The ID of the deposit.
+    function depositFor(
+        address asset,
+        uint256 amount,
+        address depositOwner
+    ) public returns (uint256) {
+        if (!supportedAssets[asset]) {
+            revert AssetNotSupported();
+        }
+        if (amount == 0) {
+            revert InvalidAmount(amount);
+        }
+        if (depositOwner == address(0) || depositOwner == asset) {
+            revert InvalidDepositOwner(depositOwner);
+        }
+
+        // Generate a new deposit ID.
+        uint256 depositId = ++depositCount;
+
+        emit DepositCreated(depositOwner, asset, depositId, amount);
+
+        // Transfer the asset from the deposit owner to the vault and approve
+        // Mezo Portal to pull them.
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
+        IERC20(asset).forceApprove(address(mezoPortal), amount);
+
+        // Create a deposit in Mezo Portal.
+        // 0 denotes no lock period for this deposit.
+        mezoPortal.deposit(address(asset), uint96(amount), 0);
```

Comment:
What is that asset is not supported in Mezo? we just trust that this asset is supported?

Reply:
If the asset is not supported in Mezo the transaction will revert with an error originated in the Mezo Portal contract.